docs/python.md

```diff
@@ -0,0 +1,207 @@
+# Python
+
+## Setup tools
+
+!!! WARNING
+    Use this only if you intend to use the system wide python,
+    which is NOT recommended.
+
+```shell
+/usr/bin/python3 -m pip install --upgrade "pip~=20.3" setuptools wheel
+```
+
+## pip configuration
+
+If you're using a private pypi server, assign its URL to `PIP_INTERNAL_INDEX_URL`
+and uncomment the `index-url` lines.
+
+If using and/or building wheels, uncomment the `[wheel]` section.
+
+```shell
+PIP_INTERNAL_INDEX_URL=
+mkdir -p "${HOME}/.config/pip"
+tee -a "${HOME}/.config/pip/pip.conf" >/dev/null <<EOF
+[global]
+# index-url = ${PIP_INTERNAL_INDEX_URL}
+# extra-index-url = https://pypi.org/simple
+no-cache-dir = true
+disable-pip-version-check = true
+
+[install]
+progress-bar = off
+
+# [wheel]
+# wheel-dir = /tmp/wheelhouse
+# find-links = /tmp/wheelhouse
+
+EOF
+```
+
+## pyenv
+
+`pyenv` handles Python version and virtual environment management.
+
+<https://github.com/pyenv/pyenv>
+
+```shell
+curl -sS -L "https://pyenv.run" | bash
+ln -s "${HOME}/.pyenv/bin/pyenv" "${HOME}/.local/bin/pyenv"
+tee -a "${HOME}/.zshrc" >/dev/null <<EOF
+
+# pyenv
+eval "\$(pyenv init --path)"
+eval "\$(pyenv init -)"
+eval "\$(pyenv virtualenv-init -)"
```

Comment:
Looks like we need to be careful to do this before loading the Oh my ZSH plugin.